### ISSUE TITLE
explicit lambda return type behavior with covariant delegates

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#73319]
I think as long as the delegate is covariant this should work since an indirect assignment of the same expression is valid.

```cs
// ok
Func<object> a1 = () => "";
        
// unexpected error
Func<object> b1 = string () => "";
```

**Comment by CyrusNajmabadi** [comment]
Definitely an interesting case!  Would have to see how the spec works here.  but i soemwhat agree.  This feels like it should fall out from lambda return types and variance.  i wonder if there's some rule preventing this, or if ti's an accidental compiler bug.

**Comment by jaredpar** [comment]
@cston is this "by design"? Not sure it's going to meet the bar for fixing right now but curious as to the intent.

**Comment by cston** [comment]
Variance is not supported for return type in the anonymous function conversion: see [Explicit return type](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md#explicit-return-type).

> Variance conversions are not allowed from lambda return type to delegate return type (matching similar behavior for parameter types).

That matches the conversion requirement for lambdas with *explicitly-typed parameters*: see [10.7 Anonymous function conversions](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/conversions.md#107-anonymous-function-conversions).

> - If `F` has an explicitly typed parameter list, each parameter in `D` has the same modifiers as the corresponding parameter in `F` and an identity conversion exists between the corresponding parameter in `F`.

Parameter example: see [sharplab.io](https://sharplab.io/#v2:C4LglgNgPgAgTARgLAChUwXAPBgDAPgAIBDBQgXkIAoB7AIwCsBTAY2EJoEoKiBvQgL4BuQgHpRhJgCcpNKYQDCAZQQA2VQhCEAdLsIAHYlOIBbJsGmFgAT31MAzoQAmNQgDsa7E8WAsAFqjomFj0zGxExHAU1KGs7Fw8hPzCqEA)

```csharp
Action<string> a1 = (object o) => { }; // error CS1661: ... parameter types do not match

Action<object> a2 = (object o) => { };
```

In the case of `a1`, the return type of the lambda expression is determined to be `object` from the target delegate type, and does not rely on variance in the anonymous function conversion.
